lib/common/metric/MetricFactory.ts

```diff
@@ -452,4 +476,58 @@ export class MetricFactory {
 
     return copy;
   }
+
+  /**
+   * Attempts to get the account from the metric if it differs from the scope.
+   */
+  private resolveAccount(
+    metricAccount: string | undefined,
+  ): string | undefined {
+    if (!this.scope) {
+      return metricAccount;
+    }
+
+    const { account } = Stack.of(this.scope);
+    if (metricAccount !== account) {
+      return metricAccount;
+    }
+
+    return;
```

Comment:
What happens if the `metricAccount` is not defined? Do we want to return `account` in those cases or we default to not passing account (which resolves to same account)?

Same question for region below.

Reply:
If they're not explicitly declared, it should just inherit what the deployment environment context is (i.e. whatever the stack is in, like what a `Metric` would normally do if you don't pass in `account` or `region`).